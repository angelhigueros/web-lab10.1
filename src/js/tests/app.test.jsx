/* eslint-disable no-undef */
import React from 'react';
import { mount } from 'enzyme';
import '@testing-library/jest-dom';

import 'jsdom-global/register';

import AppScreen from '../components/appScreen';

describe('Test en <AppScren /> ', () => {
  test('Se debe mostrar el componente correctamente', () => {
    const wrapper = mount(<AppScreen />);
    expect(wrapper).toMatchSnapshot();
  });

  test('El valor incial debe ser 0', () => {
    const wrapper = mount(<AppScreen />);
    const screen = wrapper.find('span').text();
    expect(screen).toBe('0');
  });

  test('La suma se realiza correctamente', () => {
    const wrapper = mount(<AppScreen />);
    wrapper.find('#uno').simulate('click');
    wrapper.find('#sum').simulate('click');
    wrapper.find('#dos').simulate('click');
    wrapper.find('#equal').simulate('click');
    const screen = wrapper.find('span').text();
    expect(screen).toBe('3');
  });

  test('Las operaciones concatenadas se realizan correctamente', () => {
    const wrapper = mount(<AppScreen />);
    wrapper.find('#uno').simulate('click');
    wrapper.find('#sum').simulate('click');
    wrapper.find('#dos').simulate('click');
    wrapper.find('#multi').simulate('click');
    wrapper.find('#cuatro').simulate('click');
    wrapper.find('#equal').simulate('click');
    const screen = wrapper.find('span').text();
    expect(screen).toBe('8');
  });

  test('La calculadora se reinicia correctamente', () => {
    const wrapper = mount(<AppScreen />);
    wrapper.find('#uno').simulate('click');
    wrapper.find('#sum').simulate('click');
    wrapper.find('#dos').simulate('click');
    wrapper.find('#equal').simulate('click');
    wrapper.find('#c').simulate('click');

    const screen = wrapper.find('span').text();
    expect(screen).toBe('0');
  });

  test('Los numeros se vuelven negativos correctamente', () => {
    const wrapper = mount(<AppScreen />);
    wrapper.find('#uno').simulate('click');
    wrapper.find('#sum').simulate('click');
    wrapper.find('#dos').simulate('click');
    wrapper.find('#equal').simulate('click');
    wrapper.find('#invert').simulate('click');

    const screen = wrapper.find('span').text();
    expect(screen).toBe('-3');
  });

  test('la calculadora no muestra mas de 9 digitos en pantalla', () => {
    const wrapper = mount(<AppScreen />);
    wrapper.find('#dos').simulate('click');
    wrapper.find('#dos').simulate('click');
    wrapper.find('#div').simulate('click');
    wrapper.find('#siete').simulate('click');
    wrapper.find('#equal').simulate('click');

    const screen = wrapper.find('span').text();
    expect(screen).toBe('0.3181818');
  });
});

test('El punto decimal funciona correctamente', () => {
  let wrapper = mount(<AppScreen />);
  // Caso uno - Cuenta como uno de los nueve digitos a mostrar en pantalla
  wrapper.find('#uno').simulate('click');
  wrapper.find('#dos').simulate('click');
  wrapper.find('#tres').simulate('click');
  wrapper.find('#cuatro').simulate('click');
  wrapper.find('#cinco').simulate('click');
  wrapper.find('#seis').simulate('click');
  wrapper.find('#siete').simulate('click');
  wrapper.find('#ocho').simulate('click');
  wrapper.find('#dot').simulate('click');
  wrapper.find('#nueve').simulate('click');

  const screen = wrapper.find('span').text();
  expect(screen).toBe('12345678.');

  // Caso dos - Solo se permite un punto en pantalla
  wrapper = mount(<AppScreen />);
  wrapper.find('#uno').simulate('click');
  wrapper.find('#dot').simulate('click');
  wrapper.find('#dos').simulate('click');
  wrapper.find('#dot').simulate('click');
  wrapper.find('#tres').simulate('click');

  const screen2 = wrapper.find('span').text();
  expect(screen2).toBe('1.23');
});
