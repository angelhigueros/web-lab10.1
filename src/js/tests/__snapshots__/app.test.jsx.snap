// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <AppScren />  Se debe mostrar el componente correctamente 1`] = `
<div
  className="container"
>
  <div
    className="calculator-container"
  >
    <div
      className="screen"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="row"
    >
      <div
        className="item"
        id="c"
        onClick={[Function]}
        role="button"
      >
        C
      </div>
      <div
        className="item"
        id="invert"
        onClick={[Function]}
        role="button"
      >
        +/-
      </div>
      <div
        className="item"
        id="mod"
        onClick={[Function]}
        role="button"
      >
        %
      </div>
      <div
        className="item"
        id="div"
        onClick={[Function]}
        role="button"
      >
        /
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="item"
        id="siete"
        onClick={[Function]}
        role="button"
      >
        7
      </div>
      <div
        className="item"
        id="ocho"
        onClick={[Function]}
        role="button"
      >
        8
      </div>
      <div
        className="item"
        id="nueve"
        onClick={[Function]}
        role="button"
      >
        9
      </div>
      <div
        className="item"
        id="multi"
        onClick={[Function]}
        role="button"
      >
        X
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="item"
        id="cuatro"
        onClick={[Function]}
        role="button"
      >
        4
      </div>
      <div
        className="item"
        id="cinco"
        onClick={[Function]}
        role="button"
      >
        5
      </div>
      <div
        className="item"
        id="seis"
        onClick={[Function]}
        role="button"
      >
        6
      </div>
      <div
        className="item"
        id="rest"
        onClick={[Function]}
        role="button"
      >
        -
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="item"
        id="uno"
        onClick={[Function]}
        role="button"
      >
        1
      </div>
      <div
        className="item"
        id="dos"
        onClick={[Function]}
        role="button"
      >
        2
      </div>
      <div
        className="item"
        id="tres"
        onClick={[Function]}
        role="button"
      >
        3
      </div>
      <div
        className="item"
        id="sum"
        onClick={[Function]}
        role="button"
      >
        +
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="item special"
        id="cero"
        onClick={[Function]}
        role="button"
      >
        0
      </div>
      <div
        className="item"
        id="dot"
        onClick={[Function]}
        role="button"
      >
        .
      </div>
      <div
        className="item"
        id="equal"
        onClick={[Function]}
        role="button"
      >
        =
      </div>
    </div>
  </div>
</div>
`;
